tap_googleads/custom_query_stream.py

```diff
@@ -0,0 +1,169 @@
+from types import SimpleNamespace
+from typing import Any, List, Mapping
+
+import requests
+import sqlparse
+from singer_sdk import typing as th
+
+from tap_googleads.streams import ReportsStream
+
+DATE_TYPES = ("segments.date", "segments.month", "segments.quarter", "segments.week")
+
+
+class CustomQueryStream(ReportsStream):
+    """Define custom stream."""
+
+    records_jsonpath = "$.results[*]"
+    primary_keys = []
+    replication_key = None
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Initialize the REST stream.
+
+        Args:
+            tap: Singer Tap this stream belongs to.
+        """
+        self.custom_query = kwargs.pop("custom_query")
+        self._query = self.custom_query["query"]
+        self.name = self.custom_query["name"]
+        self._schema = {}
+        super().__init__(*args, **kwargs)
+
+    @property
+    def gaql(self):
+        return self._query
+
+    def get_records(self, context):
+        self._schema = self.schema
+        yield from super().get_records(context)
+
+    @property
+    def schema(self) -> dict:
+        """Return dictionary of record schema.
+
+        Dynamically detect the json schema for the stream.
+        This is evaluated prior to any records being retrieved.
+        """
+        # Lazy evaluating so we have a customer ID to make requests with
+        if not self.context:
```

Comment:
This actually wont work. It generates the schema and sends data properly but the schema record is still using the placeholder.

Reply:
I think I may have figured it out.